massa-execution-worker/src/interface_impl.rs

```diff
@@ -253,6 +284,34 @@ impl Interface for InterfaceImpl {
         Ok(())
     }
 
+    /// Appends data to a datastore entry for the current address (top of the call stack).
+    /// Fails if the address or entry does not exist.
+    ///
+    /// # Arguments
+    /// * address: string representation of the address
+    /// * key: string key of the datastore entry
+    /// * value: value to append
+    fn raw_append_data(&self, key: &str, value: &[u8]) -> Result<()> {
+        let key = massa_hash::Hash::compute_from(key.as_bytes());
+        let mut context = context_guard!(self);
+        let addr = context.get_current_address()?;
```

Comment:
No risk of deadlock here right? I believe that the lock should be dropped after the copy but I don't remember the exact behaviour.